### PR TITLE
Upgrade to gradle 8.7

### PR DESCRIPTION
**Pull request by NickSneo**
### What does this PR do?
Upgrade to gradle 8.7

### Where should the reviewer start?
*required*

### Why is it needed?
Upgrade to gradle 8.7

## Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've added tests (if applicable).
- [ ] I've added a changelog entry if necessary.

### CODE REVIEW

File: src/main/kotlin/org/web3j/evm/ConsoleDebugTracer.kt [hunk 1]
```diff
@@ -75,7 +79,7 @@ open class ConsoleDebugTracer(protected val metaFile: File?, private val reader:
             metaFile,
             lastSourceFile,
             byteCodeContractMapping,
-            sourceFileBodyTransform
+            sourceFileBodyTransform,
```
Comment (gtebrean) [inline src/main/kotlin/org/web3j/evm/ConsoleDebugTracer.kt:82]: why we have this comma here?

File: src/main/kotlin/org/web3j/evm/EmbeddedEthereum.kt [hunk 2]
```diff
@@ -139,8 +139,8 @@ class EmbeddedEthereum(
                 @Suppress("UnstableApiUsage")
                 Resources.toString(
                     configuration.genesisFileUrl,
-                    StandardCharsets.UTF_8
-                )
+                    StandardCharsets.UTF_8,
+                ),
```
Comment (gtebrean) [inline src/main/kotlin/org/web3j/evm/EmbeddedEthereum.kt:143]: again here, is this a new compilation for kotlin?

File: src/main/kotlin/org/web3j/evm/ConsoleDebugTracer.kt (hunk 1 above)
Comment (NickSneo) [reply src/main/kotlin/org/web3j/evm/ConsoleDebugTracer.kt:82]: When I am applying spotless apply it is adding the comma there. 
Else giving error - 
`Execution failed for task ':spotlessKotlinCheck'.
> The following files had format violations:
      src/main/kotlin/org/web3j/evm/ConsoleDebugTracer.kt
          @@ -79,7 +79,7 @@
           ············metaFile,
           ············lastSourceFile,
           ············byteCodeContractMapping,
          -············sourceFileBodyTransform
          +············sourceFileBodyTransform,
           ········)
           ····}
`

File: src/main/kotlin/org/web3j/evm/EmbeddedEthereum.kt (hunk 2 above)
Comment (NickSneo) [reply src/main/kotlin/org/web3j/evm/EmbeddedEthereum.kt:143]: Seems like it is, as I have upgraded to kotlin 1.9.24 and ktlint to 0.49.1

**Review by gtebrean — APPROVED** [review]

**Pull request merged**